app/helpers/links_helper.rb

```diff
@@ -1,9 +1,12 @@
 # frozen_string_literal: true
 
+# rubocop: disable Lint/UriEscapeUnescape
+
 module LinksHelper
   def safe_slice_host(url, length = 33)
-    safe_url = sanitize(url)
+    safe_url = sanitize(URI.encode(url))
     hostname = URI.parse(safe_url).host
     truncate(hostname, length: length)
   end
 end
+# rubocop: enable Lint/UriEscapeUnescape
```

Comment:
rubocop considers URI.encode as bad practice. We need some research into why it is so. Do not merge until we get some information on this.

Reply:
Sure take your time.

Reply:
URI.encode/escape is deprecated due to certain [bugs](https://bugs.ruby-lang.org/issues/9806). Also see [this](https://stackoverflow.com/questions/2824126/whats-the-difference-between-uri-escape-and-cgi-escape) discussion.